### ISSUE TITLE
Pytest emits deprecation warning: "A private pytest class or function was used"

### ISSUE THREAD

**Issue by allanlewis** [pytest-dev/pytest-bdd#673]
I'm using `pytest-bdd` 7.0.1 and was bumping my sub-dependencies. With Pytest 8 I started to get a warning caused by this line: https://github.com/pytest-dev/pytest-bdd/blob/7.0.1/src/pytest_bdd/steps.py#L211-L218

```
pytest.PytestDeprecationWarning: A private pytest class or function was used.
```

The root cause appears to be [this change](https://github.com/pytest-dev/pytest/commit/fb55615d5e999dd44306596f340036c195428ef1#diff-49027cfd80e14edac9b0fae71f7228a408a09599f66a7815839ce8c3ae2ab84fR999) in Pytest.

**Comment by judgeaxl** [comment]
As of pytest 8.1.x this is now a hard fail as the internals have changed:
```
TypeError: FixtureDef.__init__() got an unexpected keyword argument 'fixturemanager'
```

**Comment by youtux** [comment]
Fixed in https://github.com/pytest-dev/pytest-bdd/pull/680. I just released version 7.1.2 with the fix.

Note: You will still see the warning with pytest 8.0, but if you bump to 8.1 then you shouldn't see it anymore.
